### PR TITLE
[hipsparse] Fix hipsparse compile issue

### PR DESCRIPTION
**Pull request by helloworld1**
hipsparse is not compilable due to various issues
This commit fixes:
1 pkgver reference is incorrect
2 Missing several dependencies was not set for build
3. Make install was in wrong directory

### CODE REVIEW

**Comment by acxz** [comment]
Thx for the PR! Have you been able to build this in a clean chroot?

**Comment by acxz** [comment]
Got around to trying this, now that hip-hcc works, but I get the following error: (had to remove hip>=pkgver)

```
CMake Error at /usr/share/cmake-3.17/Modules/CMakeTestCXXCompiler.cmake:53 (message):
  The C++ compiler

    "/opt/rocm/hip/bin/hipcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/acxz/vcs/git/github/helloworld1/rocm-arch/hipsparse/src/build/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/make cmTC_d2714/fast && /usr/bin/make  -f CMakeFiles/cmTC_d2714.dir/build.make CMakeFiles/cmTC_d2714.dir/build
    make[1]: Entering directory '/home/acxz/vcs/git/github/helloworld1/rocm-arch/hipsparse/src/build/CMakeFiles/CMakeTmp'
    Building CXX object CMakeFiles/cmTC_d2714.dir/testCXXCompiler.cxx.o
    /opt/rocm/hip/bin/hipcc    -march=x86-64 -mtune=generic -O2 -pipe -fno-plt    -o CMakeFiles/cmTC_d2714.dir/testCXXCompiler.cxx.o -c /home/acxz/vcs/git/github/helloworld1/rocm-arch/hipsparse/src/build/CMakeFiles/CMakeTmp/testCXXCompiler.cxx
    Linking CXX executable cmTC_d2714
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_d2714.dir/link.txt --verbose=1
    /opt/rocm/hip/bin/hipcc  -march=x86-64 -mtune=generic -O2 -pipe -fno-plt   -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now  CMakeFiles/cmTC_d2714.dir/testCXXCompiler.cxx.o  -o cmTC_d2714 
    clang-10: error: ROCm agent detector could not identify any valid targets; please specify the target explicitly by passing a valid value to -amdgpu-target
    clang-10: warning: -amdgpu-target argument 'gfx000' is not recognized; using gfx803 instead [-Winvalid-command-line-argument]
    make[1]: *** [CMakeFiles/cmTC_d2714.dir/build.make:107: cmTC_d2714] Error 1
    make[1]: Leaving directory '/home/acxz/vcs/git/github/helloworld1/rocm-arch/hipsparse/src/build/CMakeFiles/CMakeTmp'
    make: *** [Makefile:141: cmTC_d2714/fast] Error 2
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:53 (project)


-- Configuring incomplete, errors occurred!
```

**Comment by helloworld1** [comment]
Thanks for testing out. I will also test. If there a new version of hip-hcc that can be compiled on clean chroot?

**Comment by helloworld1** [comment]
I removed the `hip` from makedepend since it is already a runtime dependency. It compiles fine in clean chroot with the latest hip-hcc and other latest dependencies on AMD cpu and GPU.
It seems like only those targets are supported
@knownTargets = ('gfx701', 'gfx801', 'gfx802', 'gfx803', 'gfx900', 'gfx906', 'gfx908', 'gfx1010', 'gfx1011', 'gfx1012');

That's actually bad. The target should be runtime not compile time...

**Comment by helloworld1** [comment]
Fixed the git issue that accidentally added a bunch of files.

File: hipsparse/PKGBUILD [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Maintainer: Markus Näther <naetherm@informatik.uni-freiburg.de>
 pkgname=hipsparse
 pkgver=3.3.0
-pkgrel=2
+pkgrel=4
```
Comment (acxz) [inline hipsparse/PKGBUILD:4]: Shouldn't this be `pkgrel=3`?

**Comment by acxz** [comment]
> If there a new version of hip-hcc that can be compiled on clean chroot?
Yeah you can install hip-hcc from the arch4edu repository

File: hipsparse/PKGBUILD (hunk 1 above)
Comment (helloworld1) [reply hipsparse/PKGBUILD:4]: 3 was the previous commit that fixes compile. 4 is this commit that removes hip makedepends.

**Comment by helloworld1** [comment]
> > If there a new version of hip-hcc that can be compiled on clean chroot?
> 
> Yeah you can install hip-hcc from the arch4edu repository
> 
> > It seems like only those targets are supported
> 
> It seems like I wont be able to test this PR, I'll defer it to one of other maintainers.

Yea, I have already tested latest hip-hcc from git.

File: hipsparse/PKGBUILD (hunk 1 above)
Comment (acxz) [reply hipsparse/PKGBUILD:4]: Can you leave it at 3 since when we merge this it will squashed and merged as one commit.

Comment (helloworld1) [reply hipsparse/PKGBUILD]: Ok. Updated to 3 now.

File: hipsparse/PKGBUILD [hunk 2]
```diff
@@ -29,6 +29,9 @@ build() {
         -DBUILD_CLIENTS_SAMPLES=OFF \
         -DBUILD_CLIENTS_TESTS=OFF \
         -Drocsparse_DIR=/opt/rocm/rocsparse/lib/cmake/rocsparse \
+        -Dhip_DIR="/opt/rocm/hip/lib/cmake/hip" \
+        -Dhcc_DIR="/opt/rocm/hcc/lib/cmake/hcc" \
+        -Damd_comgr_DIR="/opt/rocm/lib/cmake/amd_comgr/" \
```
Comment (ranisalt) [inline hipsparse/PKGBUILD:34]: Don't quote these paths, and remove the trailing slash here.

**Pull request merged**